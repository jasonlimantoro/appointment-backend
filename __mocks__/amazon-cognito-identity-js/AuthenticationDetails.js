export default class AuthenticationDetails {
  constructor(authDetails = {}) {
    const { Username, Password } = authDetails;
    this.username = Username;
    this.password = Password;
  }
}
