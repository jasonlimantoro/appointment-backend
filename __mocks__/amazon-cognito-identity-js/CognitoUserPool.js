export default class CognitoUserPool {
  constructor(data = {}) {
    const { UserPoolId, ClientId } = data;
    this.userPoolId = UserPoolId;
    this.clientId = ClientId;
  }

  getCurrentUser = jest.fn().mockReturnValue('cognitouserpool');
}
