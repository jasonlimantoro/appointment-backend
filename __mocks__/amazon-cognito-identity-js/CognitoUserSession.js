export default class CognitoUserSession {
  constructor(data = {}) {
    const {
      RefreshToken = 'dummy-refreshToken',
      AccessToken = 'dummy-accessToken',
      IdToken = 'dummy-idToken',
    } = data;
    this.refreshToken = RefreshToken;
    this.accessToken = AccessToken;
    this.idToken = IdToken;
  }

  getRefreshToken = jest.fn().mockReturnValue(this.refreshToken);
}
