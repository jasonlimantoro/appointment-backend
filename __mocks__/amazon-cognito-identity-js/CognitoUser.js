import CognitoUserSession from './CognitoUserSession';

export default class CognitoUser {
  constructor(userData = {}) {
    const { Username, Pool } = userData;
    this.username = Username;
    this.pool = Pool;
  }

  getSession = jest
    .fn()
    .mockImplementation(cb => cb(null, new CognitoUserSession()));

  refreshSession = jest
    .fn()
    .mockImplementation((_refreshToken, cb) => cb(null, new CognitoUserSession()));
}
