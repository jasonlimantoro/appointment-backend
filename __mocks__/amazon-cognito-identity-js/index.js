import CognitoUser from './CognitoUser';
import CognitoUserPool from './CognitoUserPool';

module.exports = {
  CognitoUserPool: jest.fn().mockImplementation(() => new CognitoUserPool()),
  CognitoUser: jest.fn().mockImplementation(() => new CognitoUser()),
};
