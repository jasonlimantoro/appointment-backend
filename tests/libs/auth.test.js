import { CognitoUser } from 'amazon-cognito-identity-js';
import CustomAuth from '../../libs/auth';

describe('Auth', () => {
  it('refreshToken: should work', async () => {
    const res = await CustomAuth.refresh({ cognitoUsername: 'dummy-username' });
    expect(CognitoUser).toBeCalledWith(
      expect.objectContaining({ Username: 'dummy-username' }),
    );
    expect(CognitoUser).toBeCalledTimes(1);
    expect(res).toMatchSnapshot();
  });
});
