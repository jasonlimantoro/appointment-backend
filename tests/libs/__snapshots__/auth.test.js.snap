// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth refreshToken: should work 1`] = `
CognitoUserSession {
  "accessToken": "dummy-accessToken",
  "getRefreshToken": [MockFunction],
  "idToken": "dummy-idToken",
  "refreshToken": "dummy-refreshToken",
}
`;
